mutyper/cli.py

```diff
@@ -184,14 +183,16 @@ def variants(args):
             # change to propagate (can't just update indexwise)
             if variant.ploidy == 2:
                 # diploid
-                variant.genotypes = [
-                    [int(not gt[0]), int(not gt[1]), gt[2]] for gt in variant.genotypes
-                ]
+                genotype_array = variant.genotype.array()
+                # missing genotype will get converted from -1 -> 2
+                genotype_array[:, :2] = 1 - genotype_array[:, :2]
+                genotype_array[:, :2][genotype_array[:, :2] == 2] = -1
```

Comment:
I tested the timing for the two approaches and they are quite comparable, so I'll modify it in favor of readability. In addition, I switched up the code structure a little bit, because I realized the same computation could be applied to both diploid and haploid cases, let me know if you prefer the original structure and I can switch it back. Also added the genotype element check.